app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,27 @@
+class PasswordResetsController < ApplicationController
+
+  def show
+    user = User.where(token: params[:id]).first
+    if user
+      @token = user.token 
+    else 
+      redirect_to expired_token_path 
+    end
+  end
+
+  def expired_token
+  end
+
+  def create
+    user = User.where(token: params[:token]).first
+    if user 
+      user.password = params[:password]
+      user.save
```

Comment:
I am not sure about this. So if the password is not valid then it breaks right. This is not correctly handled.